### PR TITLE
Add missing optional prop to forms

### PR DESCRIPTION
**Pull request by digitalsadhu**
Fixes https://github.com/fabric-ds/issues/issues/109

There's a fair amount of formatting which is completely contained in the commit aeb2c4041b8e1abdc89a58cdeb5bc46759cf6f93 so for PR review it is probably easier to look at just b6415d22bffcb98dd58ba9445749fccab674cc4f

### CODE REVIEW

**Review by pearofducks — APPROVED** [review]
lgtm!

File: packages/textarea/docs/TextArea.mdx [hunk 1]
```diff
@@ -56,6 +56,14 @@ TextArea for accessibility. If the field is labeled by a separate element, an
 `aria-labelledby` prop must be provided using the id of the labeling element
 instead.
 
+### Optional
+
+Add the optional prop to indicate that the textarea is not required.
+
+```jsx example
+<TextField label="Description" optiona />
```
Comment (BalbinaK) [inline packages/textarea/docs/TextArea.mdx:64]: It should be `<TextArea label="Description" optional />`

File: packages/select/src/component.tsx [hunk 2]
```diff
@@ -45,11 +56,21 @@ const setup = (props) => {
...
 
   return (
     <div className={classes} {...div}>
-      {label.children && <label {...label} />}
+      {label.children && (
+        <label htmlFor={label.htmlFor}>
+          {label.children}
+          {optional && (
+            <span className="pl-8 font-normal text-14 text-gray-500">
+              {' '}
```
Comment (BalbinaK) [inline packages/select/src/component.tsx:68]: Why do we need to both set left padding and add an empty space here?

File: packages/select/src/component.tsx [hunk 3]
```diff
@@ -4,7 +4,17 @@ import { classNames } from '@chbphone55/classnames';
...
+    className,
+    invalid,
+    id,
+    hint,
+    always,
+    label,
+    style,
+    optional,
+    ...rest
+  } = props;
```
Comment (BalbinaK) [inline packages/select/src/component.tsx:17]: Should we also add `optional` type to `SelectProps`?

File: packages/textarea/docs/TextArea.mdx (hunk 1 above)
Comment (digitalsadhu) [reply packages/textarea/docs/TextArea.mdx:64]: whoops!

File: packages/select/src/component.tsx (hunk 3 above)
Comment (digitalsadhu) [reply packages/select/src/component.tsx:17]: nice catch, had forgotten to save the file locally

File: packages/select/src/component.tsx (hunk 2 above)
Comment (digitalsadhu) [reply packages/select/src/component.tsx]: another nice catch, totally can't see a reason for that, removing.

**Review by BalbinaK — APPROVED** [review]
🎉

**Pull request merged**